app/src/main/kotlin/com/hippo/ehviewer/image/Image.kt

```diff
@@ -112,9 +114,7 @@ class Image private constructor(image: CoilImage, private val src: AutoCloseable
                 if (isAtLeastO) {
                     colorSpace(colorSpace)
                 }
-                if (Settings.cropBorder.value) {
-                    allowHardware(false)
-                }
+                allowHardware(false)
```

Comment:
For non-crop scenarios, we always use a non-hw bitmap, which decreases performance

Reply:
Well, we have no information about image size so can't disable hardware bitmap based on that.

Reply:
> Well, we have no information about image size so can't disable hardware bitmap based on that.

Make non-hw decode a fallback path

Reply:
> For non-crop scenarios, we always use a non-hw bitmap, which decreases performance

Wait didn't we copy it to hardware bitmap regardless of crop borders settings?

Reply:
> > For non-crop scenarios, we always use a non-hw bitmap, which decreases performance
> 
> Wait didn't we copy it to hardware bitmap regardless of crop borders settings?

Oops, that's terrible, we need to fix it up

Reply:
Let's add a Interceptor to address the fallback path
check hardware flags first, if true, check the result, if is that Exception then we re-decode it